src/workloads/query/ProjectParse.yml

```diff
@@ -0,0 +1,91 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  Test $project parsing performance for small and large number of fields.
+
+GlobalDefaults:
+  dbname: &db test
+  coll: &coll Collection0
+  nop: &Nop {Nop: true}
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10
+    BatchSize: 10
+    Document:
+      m: {^Inc: {start: 0}}
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - *Nop
+
+- Name: Queries
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 10000
+    Database: *db
+    Operations:
+    - OperationMetricsName: ProjectParseSmall
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: *coll
+        pipeline: [
+          {$project : {^Object: {
+              withNEntries: 10,
+              havingKeys: {^RandomString: {length: 10}},
+              andValues: 1,
+              duplicatedKeys: skip
+            }}}
+        ]
+        cursor: {}
+  - Repeat: 50
+    Database: *db
+    Operations:
+    - OperationMetricsName: ProjectParseLarge
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: *coll
+        pipeline: [
+          {$project : {^Object: {
+              withNEntries: 20000,
+              havingKeys: {^RandomString: {length: 10, alphabet: "abcdefghijklmnopqrstuvwxyz"}},
```

Comment:
You might want to use FastRandomString here. It's a lower quality number generator, but faster.

I have some concern that you are going to saturate the workload client generating the random data for this pipeline. You could compare the average latency to the throughput on this one to see if they match up. I think data generation time should not show up in the average latency, so you can check if OperationThroughput = 1/Average Latency.

Reply:
What if we wrapped this with ^FixedGeneratedValue? I think that would be a good workaround.

Reply:
Where would you put the FixedGeneratedValue? Would you wrap the whole ^Object? If you did that the first execution still needs to generate all 20k entries and could still be slow. Would that be good enough? I don't think you can wrap the RandomString call itself, because then you would get 20k identical keys.

Reply:
I was thinking around the ^Object would be good. Maybe both would be beneficial. The first run would create 20k fields using ^FastRandomString and reuse that result for the rest of the runs.